gen/calc.py

```diff
@@ -419,6 +429,68 @@ def calculate_cluster_docker_registry_enabled(cluster_docker_registry_url):
     return 'false' if cluster_docker_registry_url == '' else 'true'
 
 
+def validate_cosmos_config(cosmos_config):
+    """The schema for this configuration is.
+    {
+      "schema": "http://json-schema.org/draft-04/schema#",
+      "type": "object",
+      "properties": {
+        "staged_package_storage_uri": {
+          "type": "string"
+        },
+        "package_storage_uri": {
+          "type": "string"
+        }
+      }
+    }
+    """
+    config = validate_json_dictionary(cosmos_config)
+    if ('staged_package_storage_uri' in config and
```

Comment:
How about something like the following? I think it simplifies the code a bit, makes it harder to get wrong.

```
expected = ['staged_package_storage_uri', 'package_storage_uri']
found = list(filter(lambda param: param in config, expected))
if len(found) == 0:
    return

if len(found) == 1:
    raise AssertionError(
        'cosmos_config must be a dictionary containing both {}, or must be left '
        'empty. Found only {}'.format(' '.join(expected), found))

for parameter in expected:
    validate_url(config[parameter])
```